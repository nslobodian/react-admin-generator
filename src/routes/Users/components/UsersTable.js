import React from 'react'
import Field from 'redux-form/es/Field'
import reduxForm from 'redux-form/es/reduxForm'
import pure from 'recompose/pure'

const areas = {
  igv: <label className='label label-success'>IGV</label>,
}

const program_statuses = {
  completed: <label className='label label-info'>Completed</label>,
}

const ldm_statuses = {
  false: <label className='label label-danger'>Nope</label>,
}

export const UsersTable = pure(({ users, handleChangeFilter, goToUserDetails }) =>
  <div className='col-xs-12'>
    <div className='table-responsive'>
      <table className='table table-hover'>
        <thead>
          <tr>
            <td>EXPA ID</td>
            <td>Full Name</td>
            <td>Position</td>
            <td>Area</td>
            <td>IXP</td>
            <td>LDM</td>
            <td />
          </tr>
          <tr>
            {
              fields.map(field =>
                <td key={field.value}>
                  <Field
                    component='input'
                    className='form-control'
                    name={field.value}
                    placeholder={field.name}
                    onChange={(pros, value) => handleChangeFilter(field.value, value)}
                  />
                </td>
              )
            }
            <td />
          </tr>
        </thead>
        <tbody>
          {users.map(({ expa_id, surname, name, position, area, ixp, ldm, id }) =>
            <tr key={id}>
              <td>{expa_id}</td>
              <td>{`${name} ${surname}`}</td>
              <td>{position}</td>
              <td>{areas[area]}</td>
              <td>{program_statuses[ixp]}</td>
              <td>{ldm_statuses[ldm]}</td>
              <td>
                <button className='btn btn-primary' onClick={() => goToUserDetails(id)}>View</button>
              </td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  </div>
)

const fields = [
  { name: 'Expa Id', value: 'expa_id' },
  { name: 'Full Name', value: 'full_name' },
  { name: 'Position', value: 'position' },
  { name: 'Area', value: 'area' },
  { name: 'IXP', value: 'ixp' },
  { name: 'LDM', value: 'ldm' },
]

export default reduxForm({
  form: 'UsersTable',
})(UsersTable)
