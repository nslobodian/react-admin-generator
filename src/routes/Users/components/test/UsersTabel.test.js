import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { UsersTable } from '../UsersTable'

const props = {
  fields: [
    {
      value: 1,
      name: 'name',
    },
  ],
  users: [{
    username: 'Username',
    url: 'google.com',
    lastName: 'LastName',
    name: 'name',
    email: 'email@dot.com',
    userId: 1234,
  }],
  handleChangeFilter: jest.fn(),
  goToUserDetails: jest.fn(),
}

describe('(Route) UsersTable', () => {
  it('Should render component properly', () => {
    const wrapper = shallow(
      <UsersTable
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Component button click', () => {
    const wrapper = shallow(
      <UsersTable
        {...props}
      />)
    wrapper.find('button').simulate('click')
  })
})
