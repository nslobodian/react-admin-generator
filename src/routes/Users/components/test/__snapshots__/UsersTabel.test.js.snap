// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) UsersTable Should render component properly 1`] = `
<div
  className="col-xs-12"
>
  <div
    className="table-responsive"
  >
    <table
      className="table table-hover"
    >
      <thead>
        <tr>
          <td>
            Username
          </td>
          <td>
            First Name
          </td>
          <td>
            Last Name
          </td>
          <td>
            Email
          </td>
          <td>
            Url
          </td>
          <td />
        </tr>
        <tr>
          <td>
            <Field
              className="form-control"
              component="input"
              name="username"
              onChange={[Function]}
              placeholder="User Name"
            />
          </td>
          <td>
            <Field
              className="form-control"
              component="input"
              name="name"
              onChange={[Function]}
              placeholder="First Name"
            />
          </td>
          <td>
            <Field
              className="form-control"
              component="input"
              name="lastName"
              onChange={[Function]}
              placeholder="Last Name"
            />
          </td>
          <td>
            <Field
              className="form-control"
              component="input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
            />
          </td>
          <td>
            <Field
              className="form-control"
              component="input"
              name="url"
              onChange={[Function]}
              placeholder="Url"
            />
          </td>
          <td />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Username
          </td>
          <td>
            name
          </td>
          <td>
            LastName
          </td>
          <td>
            <a
              href="email@dot.com"
            >
              email@dot.com
            </a>
          </td>
          <td>
            <a
              href="//herlandia.com/google.com"
              target="blank"
            >
              herlandia.com/google.com
            </a>
          </td>
          <td>
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Details
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
